src/unix/aix.c

```
@@ -128,8 +498,390 @@ void uv_loadavg(double avg[3]) {
 }
 
 
+static char *uv__rawname(char *cp) {
+  static char rawbuf[FILENAME_MAX+1];
+  char *dp = rindex(cp, '/');
+
+  if (dp == 0)
+    return (0);
+
+  *dp = 0;
+  strcpy(rawbuf, cp);
+  *dp = '/';
+  strcat(rawbuf, "/r");
+  strcat(rawbuf, dp+1);
+  return (rawbuf);
+}
+
+
+/* 
+ * Determine whether given pathname pertains to a file or a directory
+ * Returns 0 for file, 1 for directory, or an error code < 0 on failure
+ *
+ * TODO: most callers of this don't check the error case
+ */
+static int uv__file_is_a_directory(char* filename) {
+  struct stat statbuf;
+  int rc = 0;
+
+  if (stat(filename, &statbuf) < 0) {
+    return -errno;
+  }
+  if (statbuf.st_type == VDIR) {
+    rc = 1;
```

Actually, this function is a bit weird: it returns -errno for failure, 1 for success and 0 for other types of failure. Since it's called `uv__file_is_a_directory` i'd go for returning 1 on sucess and 0 on failure, swallowing the error.